src/tree.service.ts

```diff
@@ -24,6 +24,8 @@ export class TreeService {
   public nodeExpanded$: Subject<NodeExpandedEvent> = new Subject<NodeExpandedEvent>();
   public nodeCollapsed$: Subject<NodeCollapsedEvent> = new Subject<NodeCollapsedEvent>();
 
+  public APIs: Object = { };
```

Comment:
you can use `Map` class in here - it has more convenient API and allows you to have objects as keys. Also we should have a look at WeakMap, in case if some of the nodes are deleted